lib/Base/Organization/CollectionSorters.vala

```diff
@@ -0,0 +1,98 @@
+/* CollectionSorters.vala
+ *
+ * Copyright 2023 Frederick Schenk
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+using GLib;
+
+/**
+ * Interface for methods used to sort collections.
+ */
+public interface Backend.CollectionSorter <T> : Object {
+
+  /**
+   * Compares two items while sorting the list.
+   *
+   * @param a The first item to compare.
+   * @param b The second item to compare.
+   *
+   * @return How the items are sorted (positive when a before b, negative when b before a).
+   */
+  public abstract int compare_items (T a, T b);
+
+}
+
+/**
+ * Sorts a collection with posts reverse chronological.
+ *
+ * The sorting is mostly reverse chronological, with a few exceptions.
+ * Any item that is not a Post, particular PseudoItems, will be placed on top.
+ */
+public class Backend.ReversePostSorter : CollectionSorter <Object>, Object {
+
+  /**
+   * Compares two items while sorting the list.
+   *
+   * @param a The first item to compare.
+   * @param b The second item to compare.
+   *
+   * @return How the items are sorted (positive when a before b, negative when b before a).
+   */
+  protected int compare_items (Object a, Object b) {
+    // Attempt to cast as Posts
+    var post_a = a as Post;
+    var post_b = b as Post;
+    if (a == null || b == null) {
+      return compare_non_posts (a, b);
+    }
```

Comment:
```suggestion
  protected int compare_items (Object a, Object b) {
    // Attempt to cast as Posts
    var post_a = a as Post;
    var post_b = b as Post;
    if (post_a == null || post_b == null) {
      return compare_non_posts (a, b);
    }
```
Would that fix the segfault? Because I _think_ it's currently doing dynamic type casting but then still checking the originals for `null`-ness, so a pair of non-`null` non-`Post` objects would have `post_a = null; post_b = null;` but then not match the IF and go on to try and get the creation date from a non-`Post` object.

Reply:
I guess there was my mistake. Need to commit that change when I'm back at my computer...

Reply:
While I had made a mistake there, it wasn't the one to cause the segfault.

It already fails at the `post_a = a as Post` line. I have no idea why...